### PR TITLE
Begin `hop ignite from-compose`

### PR DESCRIPTION
**Pull request by alii**
I am learning rust

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -4,8 +4,6 @@ use anyhow::Result;
 use clap::Parser;
 use hop_cli::commands::handle_command;
 use hop_cli::commands::update::util::version_notice;
-#[cfg(feature = "update")]
-use hop_cli::commands::Commands;
```
Comment (pxseu) [inline src/main.rs]: what are these two line removes for?

Comment (alii) [reply src/main.rs]: didnt like it

Comment (pxseu) [reply src/main.rs]: kk

File: src/commands/ignite/from_compose.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    let compose: DockerCompose = match serde_yaml::from_str(&compose) {
+        Ok(compose) => compose,
+        Err(error) => {
+            println!("{:?}", error.location());
+
+            return Err(Error::new(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                format!("Failed to parse {}: {}", file, error),
+            )));
```
Comment (pxseu) [inline src/commands/ignite/from_compose.rs]: this too

File: Cargo.toml [hunk 3]
```diff
@@ -51,6 +56,7 @@ hyper = { version = "0.14.20", features = ["server"] }
 ctrlc = { version = "3.2.2", features = ["termination"] }
 chrono = { version = "0.4.21", features = ["serde"] }
 async-compression = { version = "0.3.14", features = ["tokio", "gzip"] }
+serde_derive = "1.0.144"
```
Comment (pxseu) [inline Cargo.toml]: remove this dep since derive is part of serde now

**Review by pxseu — CHANGES_REQUESTED** [review]

File: src/commands/ignite/from_compose.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    let default_file = "docker-compose.yml".to_owned();
+    let file = options.file.unwrap_or(default_file);
+
+    let path = Path::new(&file);
+
+    if !path.exists() {
+        return Err(Error::new(std::io::Error::new(
+            std::io::ErrorKind::NotFound,
+            format!("File {} does not exist", file),
+        )));
```
Comment (pxseu) [inline src/commands/ignite/from_compose.rs]: replace for anyhow macro

**Review by pxseu — DISMISSED** [review]
LGTM ALI🔥

**Review by pxseu — APPROVED** [review]

**Pull request merged**